test/src/test/kotlin/graphql/nadel/tests/hooks/service-context-is-being-set.kt

```diff
@@ -0,0 +1,41 @@
+package graphql.nadel.tests.hooks
+
+import graphql.nadel.Nadel
+import graphql.nadel.ServiceExecution
+import graphql.nadel.ServiceExecutionFactory
+import graphql.nadel.hooks.CreateServiceContextParams
+import graphql.nadel.hooks.HooksVisitArgumentValueEnvironment
+import graphql.nadel.hooks.NewVariableValue
+import graphql.nadel.hooks.ServiceExecutionHooks
+import graphql.nadel.tests.EngineTestHook
+import graphql.nadel.tests.KeepHook
+import graphql.nadel.tests.NadelEngineType
+import graphql.nadel.tests.util.serviceExecutionFactory
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+import java.util.concurrent.CompletableFuture
+
+@KeepHook
+class `service-context-is-being-set` : EngineTestHook {
+    override fun makeNadel(engineType: NadelEngineType, builder: Nadel.Builder): Nadel.Builder {
+        val serviceExecutionFactory = builder.serviceExecutionFactory
+
+        return builder
+            .serviceExecutionHooks(object : ServiceExecutionHooks {
+                override fun createServiceContext(params: CreateServiceContextParams?): CompletableFuture<Any?>? {
```

Comment:
Yea this is the annoying thing about Java compat, it just assumes this is nullable because there's no nullability annotations 😢